src/librustc_builtin_macros/format.rs

```diff
@@ -894,7 +894,7 @@ pub fn expand_preparsed_format_args(
     };
 
     let (is_literal, fmt_snippet) = match ecx.source_map().span_to_snippet(fmt_sp) {
-        Ok(s) => (s.starts_with("\"") || s.starts_with("r#"), Some(s)),
+        Ok(s) => (s.starts_with('\"') || s.starts_with("r#"), Some(s)),
```

Comment:
No need to escape this.
```suggestion
        Ok(s) => (s.starts_with('"') || s.starts_with("r#"), Some(s)),
```